scripts/pre_commit_linter.py

```diff
@@ -1312,6 +1336,84 @@ def _check_js_component_name_and_count(self):
             print ''
             return summary_messages
 
+    def check_for_mandatory_pattern_in_file(
+            self, pattern_set, filepath, failed):
+        """Checks for a given mandatory pattern in a file.
+
+        Args:
+            pattern_set: list(dict). The mandatory pattern to be checked in the
+                file.
+            filepath: str. The path to the file to be linted.
+            failed: bool. Status of failure of the check.
+
+        Returns:
+            bool. The failure status of the check.
+        """
+        # This boolean list keeps track of the regex matches
+        # found in the file.
+        pattern_found_list = []
+        file_content = FileCache.readlines(filepath)
+        for index, regexp_to_check in enumerate(
+                pattern_set):
+            if (any([filepath.endswith(
+                    allowed_type) for allowed_type in (
+                        regexp_to_check['included_types'])]) and (
+                            not any([
+                                filepath.endswith(
+                                    pattern) for pattern in (
+                                        regexp_to_check[
+                                            'excluded_files'] +
+                                        regexp_to_check[
+                                            'excluded_dirs'])]))):
+                pattern_found_list.append(False)
+                for line in file_content:
+                    if re.search(regexp_to_check['regexp'], line):
+                        pattern_found_list[index] = True
+                        break
+        if not all(pattern_found_list):
+            failed = True
+            for index, pattern_found in enumerate(
+                    pattern_found_list):
+                if not pattern_found:
+                    print '%s --> %s' % (
+                        filepath,
+                        pattern_set[index]['message'])
+        return failed
+
+    def _check_mandatory_patterns(self):
+        """This function checks that all files contain the mandatory
+        patterns.
+        """
+        if self.verbose_mode_enabled:
+            print 'Starting mandatory patterns check'
+            print '----------------------------------------'
+
+        summary_messages = []
+        failed = False
+
+        with _redirect_stdout(_TARGET_STDOUT):
+            sets_of_patterns_to_match = [
```

Comment:
`sets` --> `set`

Reply:
Actually the list is a list of lists of sets of patterns which are mandatory/JS specific and hence the name. Please let me know if this is a mistake.